### DISCUSSION TITLE
Implementing F0 mean filtering

### DISCUSSION THREAD

**Discussion by tovaru** [voicepaw/so-vits-svc-fork#251] [Ideas]
The original repository added this new feature that greatly improves audio quality for songs with a lot of harmonics and reverb. I think it would be great to have it here, would make things much easier for getting accurate voices on songs.

**Comment by BlueAmulet** [comment]
Relevant commits
https://github.com/svc-develop-team/so-vits-svc/commit/9eb0525ad8b4cfcb75a39d2a2b57adede299dd69
https://github.com/svc-develop-team/so-vits-svc/commit/f25e5667d72468aa6428a2d0b4f70f74d3ff5944

Note: This doesn't seem to actually be of much besides a selector between parselmouth and crepe.

**Comment by Lordmau5** [comment]
On first glance, it seems like this project already does handle separate F0 methods / calculations

Namely, "pyworld" for dio and harvest, "crepe" for crepe and crepe-tiny, and "parselmouth" for parselmouth

https://github.com/34j/so-vits-svc-fork/blob/main/src/so_vits_svc_fork/f0.py#L188

I am, however, not sure if these can be improved with the changes in those commits

**Comment by 34j** [comment]
As usual, I'm not sure what the upstream repo code is doing.

Why not simply median().bfill()?
